camayoc/tests/qpc/cli/test_openshift.py

```diff
@@ -116,10 +112,22 @@ def validate_openshift_with_acm(cluster, details):
                     assert int(metric["socket_worker"]) >= 0
 
 
-@pytest.mark.parametrize(
-    "cluster", config_openshift(), ids=(x["hostname"] for x in config_openshift())
-)
-def test_openshift_clusters(cluster, qpc_server_config):
+def openshift_sources():
+    for source_definition in settings.sources:
+        if source_definition.type != "openshift":
+            continue
+        fixture_id = source_definition.name
+        yield pytest.param(source_definition, id=fixture_id)
+
+
+def openshift_cluster_info(name):
+    for scan_info in settings.scans:
+        if name == scan_info.name:
```

Comment:
This function will receive `source_definition.name` as an input, and this condition will only pass if scan happens to be named the same as source.

I know we have full control over configuration file and we are generally not very strict about enforcing some rules, but I think this is taking it a little too far. The correct solution is to go through scans and check if scan depends on source with a name passed as an argument.

But what if one source is used in multiple scans?

I think it would be easier to run `test_openshift_clusters` for each scan, not source. Then even if you have multiple scans depending on single source, they will both work. And since we enforce scan names uniqueness when reading settings file, you can take a source name from scan definition and loop over all sources, looking for one with matching name.

The only problem that would remain is that technically, single scan may have multiple sources. I don't think trying to scan both openshift and network makes much sense, but I think we do allow to scan multiple openshift sources. However, I have not tried it and we can afford to worry about it later.

Reply:
After reading your comments here is some thoughs I have to address the scans and its data verification.  Hope we can align something for a new PR.

1. Instead of creating CLI scans in the test itself, transform these scans as pytest test fixtures and put them in `conftest.py`. I believe that the scan information would be cached and would make inspections less time consuming. And we can walk through the scans to find what to verify with a defined source. I think it is like: *The correct solution is to go through scans and check if scan depends on source with a name passed as an argument*
2. Make use (in some way similar to what we have for creds, sources, etc.) of a Pydantic model for the expected data information and use the validation functions to check if the data is correct, instead of just using the Python conditionals and sifting through data dictionaries.
3. (Maybe) Separate/detach the expected data part from the scans. Creating a "verification/reports/expected-data" session especially for this type of validation. The logic I'm thinking here is to have validation for one particular expected data report to one or more scans.